plugins/value-container-assignment/testData/diagnostics/otherOperators.kt

```diff
@@ -0,0 +1,71 @@
+annotation class ValueContainer
+
+@ValueContainer
+data class StringProperty(var v: String) {
+    fun assign(v: String) {
+        this.v = v
+    }
+    fun assign(v: StringProperty) {
+        this.v = v.get()
+    }
+    fun get(): String = v
+}
+
+data class Task(val input: StringProperty)
+
+fun `should not effect error reporting for other operators`() {
+    val task = Task(StringProperty("Fail"))
+    val nullTask: Task? = null
+
+    // a.b += c
+    task.input <!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>+=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>+=<!> StringProperty("Fail")
+
+    // a.b -= c
+    task.input <!UNRESOLVED_REFERENCE!>-=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE!>-=<!> StringProperty("Fail")
+
+    // a.b *= c
+    task.input <!UNRESOLVED_REFERENCE!>*=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE!>*=<!> StringProperty("Fail")
+
+    // a.b /= c
+    task.input <!UNRESOLVED_REFERENCE!>/=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE!>/=<!> StringProperty("Fail")
+
+    // a.b %= c
+    task.input <!UNRESOLVED_REFERENCE!>%=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE!>%=<!> StringProperty("Fail")
+
+    // a.b <= c
+    task.input <!UNRESOLVED_REFERENCE!><=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE!><=<!> StringProperty("Fail")
+
+    // a.b >= c
+    task.input <!UNRESOLVED_REFERENCE!>>=<!> StringProperty("Fail")
+    nullTask?.input <!UNRESOLVED_REFERENCE!>>=<!> StringProperty("Fail")
+
+    // a.b[c] = d
+    task.input<!NO_SET_METHOD!><!UNRESOLVED_REFERENCE!>[<!>0<!UNRESOLVED_REFERENCE!>]<!><!> = StringProperty("Fail")
+    nullTask?.input<!NO_SET_METHOD!><!UNRESOLVED_REFERENCE!>[<!>0<!UNRESOLVED_REFERENCE!>]<!><!> = StringProperty("Fail")
+
+    // a.b[c, d] = e
+    task.input<!NO_SET_METHOD!><!UNRESOLVED_REFERENCE!>[<!>0, 0<!UNRESOLVED_REFERENCE!>]<!><!> = StringProperty("Fail")
+    nullTask?.input<!NO_SET_METHOD!><!UNRESOLVED_REFERENCE!>[<!>0, 0<!UNRESOLVED_REFERENCE!>]<!><!> = StringProperty("Fail")
+
+    // a.b[c,..,d] = e
+    task.input<!NO_SET_METHOD!><!UNRESOLVED_REFERENCE!>[<!>0, 0, 0<!UNRESOLVED_REFERENCE!>]<!><!> = StringProperty("Fail")
+    nullTask?.input<!NO_SET_METHOD!><!UNRESOLVED_REFERENCE!>[<!>0, 0, 0<!UNRESOLVED_REFERENCE!>]<!><!> = StringProperty("Fail")
+
+    // a?.b[c] += d
```

Comment:
What about `a?.b = c`?

Reply:
Added and fixed resolution for old front-end.